### PR TITLE
refactor: [M3-7230] - Migrate NodeBalancer Node Mode Select TextField to Autocomplete

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Very similar PR to https://github.com/linode/manager/pull/9752 (Only 1 more to go after this one 😁)
- In order to fix up our TextField component, we need to not allow it to be used as a select component. This PR replaces the TextField Select in the NodeBalancer config node component with a modern alternative like Autocomplete

## Major Changes 🔄
- Clean up `Remove` button styles
- Replace `<TextField select>...</TextField>` with an `<Autocomplete />`
- Removes some unused testing attributes

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-10-04 at 12 28 13 PM | Screenshot 2023-10-04 at 12 27 57 PM |

## How to test 🧪
- Deploy a Linode with a private ipv4
- Go create a NodeBalancer in the same region as that Linode
- Add your Linode as a node and complete the NodeBalancer creation 
- Go to that NodeBalancer's config and test the `Mode` Select
  - Test styles 🎨
  - Test functionality  🎮

### CODE REVIEW

**Review by hana-linode — CHANGES_REQUESTED** [review]
Looks like removing a Backend Node broke 🤔

https://github.com/linode/manager/assets/115299789/4767f9c6-7acc-4135-a0b5-ce084c2c359b

**Comment by bnussman-akamai** [comment]
@hana-linode Thank you! Fixed in eb9667a

File: packages/manager/src/features/NodeBalancers/NodeBalancerConfigNode.tsx [hunk 1]
```diff
@@ -59,6 +59,16 @@ export const NodeBalancerConfigNode = React.memo(
...
 
+    const modeOptions: {
+      label: string;
+      value: NodeBalancerConfigNodeMode;
+    }[] = [
+      { label: 'Accept', value: 'accept' },
+      { label: 'Reject', value: 'reject' },
+      { label: 'Backup', value: 'backup' },
+      { label: 'Drain', value: 'drain' },
+    ];
```
Comment (hana-linode) [inline packages/manager/src/features/NodeBalancers/NodeBalancerConfigNode.tsx:70]: We can move this outside of the component

**Review by hana-linode — APPROVED** [review]
Confirmed that the `Mode` select still looks and works as expected and that remove still works 🎉

**Review by jdamore-linode — APPROVED** [review]
Thanks @bnussman-akamai! Confirmed that the autocomplete field works/looks as expected and that I can add/remove nodes. Also confirmed that I can actually save the config changes

**Pull request merged**